tools/convert-shared-credential-to-legacy-format.rb

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env ruby
+
+require 'json'
+
+tools_dir = __dir__
```

Comment:
More for my information than a review comment, in Ruby is `__dir__` the actual fully-qualified directory of the script? I believe in Python (and/or Bash?) it can just be `.` which can cause problems if you rely on it to figure out paths.

Reply:
Oh my gosh: this is amazing. https://docs.ruby-lang.org/en/2.4.0/Kernel.html#method-i-__dir__

Reply:
(Please don't take this surprise and delight as endorsement of Ruby. 😉)